lang/src/org/partiql/lang/syntax/SqlParser.kt

```diff
@@ -1415,10 +1374,28 @@ class SqlParser(private val ion: IonSystem) : Parser {
 
             else -> ParseNode(TYPE, head, emptyList(), tail)
         }
-        // Check for the optional time zone for TIME
+        // Check for the optional "WITH TIME ZONE" specifier for TIME and validate the value of precision.
+        // Note that this needs to be checked explicitly as the keywordtext for "TIME WITH TIME ZONE" consists of multiple words.
```

Comment:
Do you think this should be in the lexer?  Or rather, did we sort that out in another PR?

Reply:
We discussed about having a keyword associated with `TIME WITH TIME ZONE` in the lexer. That is doable. However, I don't think we can model `TIME (<precision>) WITH TIME ZONE` in the lexer DFA.
Another option was to model just `WITH TIME ZONE` in the lexer but note that `WITH TIME ZONE` on itself shouldn't be considered as a keyword (or a series of keywords).
Considering all these, I thought it wasn't possible to model this in lexer DFA.